server/src/main/java/org/apache/druid/server/http/SegmentListerResource.java

```diff
@@ -327,6 +330,25 @@ public void onFailure(Throwable th)
     asyncContext.setTimeout(timeout);
   }
 
+  @GET
+  @Path("/segmentLoadingCapabilities")
+  @Produces({MediaType.APPLICATION_JSON, SmileMediaTypes.APPLICATION_JACKSON_SMILE})
+  public Response getSegmentLoadingCapabilities(
+      @Context final HttpServletRequest req
+  ) throws IOException
+  {
+    if (loadDropRequestHandler == null) {
+      sendErrorResponse(req, HttpServletResponse.SC_NOT_FOUND, "load/drop handler is not available.");
```

Comment:
I am not sure if this is needed since you are returning a response in the next line itself.